test/integration/bodyArray.spec.ts

```diff
@@ -601,9 +601,8 @@ describe("Integration tests for BodyArrayClient", () => {
     });
   });
 
-  describe.skip("#getBooleanInvalidString", () => {
+  describe("#getBooleanInvalidString", () => {
     it("returns [true, 'boolean', false]", async () => {
-      // TODO: Should throw deserialization error
```

Comment:
Are these skipped tests supposed to be throwing errors? (I think I might have left those TODOs originally). If not then nevermind!

Reply:
Discussed offline. If we want to guard against arrays with elements that don't match what is defined in the type, for example here, `getBooleanInvalidString's` return type is an `Array<boolean>` but TestServer sends an array with booleans and a string.

This would need to be verified at runtime, since these arrays are valid in Javascript.

I've created an issue in `core-http` to discuss implementing this